### PR TITLE
add modules for managing facility network and hardware

### PR DESCRIPTION
**Pull request by saladgg**

### CODE REVIEW

File: fahari/mle/models.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+from fahari.common.models import AbstractBase, Facility
+
+
+class FacilityNetworkCondition(AbstractBase):
+    """Facility network conditions."""
+
+    facility = models.ForeignKey(Facility, on_delete=models.PROTECT)
+    TRUE = "TRUE"
+    FALSE = "FALSE"
+    STATE_INDICATORS = [
```
Comment (kennedykori) [inline fahari/mle/models.py:14]: Use choice enumeration types instead here. Check the [docs](https://docs.djangoproject.com/en/3.2/ref/models/fields/#enumeration-types).

File: fahari/mle/models.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        (TRUE, "True"),
+        (FALSE, "False"),
+    ]
+    has_network = models.CharField(max_length=50, choices=STATE_INDICATORS, default=TRUE)
```
Comment (kennedykori) [inline fahari/mle/models.py:18]: Actually this should be a boolean field, not a choice field.

File: fahari/mle/models.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    has_internet = models.CharField(max_length=50, choices=STATE_INDICATORS, default=TRUE)
```
Comment (kennedykori) [inline fahari/mle/models.py:19]: This too. Change this to a boolean field.

File: fahari/mle/models.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+    class Meta:
+        ordering = ("facility__name", "-updated")
+
+
+class FacilityDeviceRequest(AbstractBase):
+    """Facility devices request by users."""
+
+    facility = models.ForeignKey(Facility, on_delete=models.PROTECT)
+    device_requested = models.CharField(max_length=50, default="")
```
Comment (kennedykori) [inline fahari/mle/models.py:63]: What is this field supposed to capture? The name of the device?

File: fahari/mle/models.py [hunk 5]
```diff
@@ -0,0 +1,85 @@
+from django.contrib.gis.db import models
+from django.urls.base import reverse
+from django.utils import timezone
+
+from fahari.common.models import AbstractBase, Facility
+
+
+class FacilityNetworkCondition(AbstractBase):
```
Comment (kennedykori) [inline fahari/mle/models.py:8]: Could you change this name to something more descriptive? What is a network condition?

File: fahari/mle/models.py [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+            "-updated",
+            "facility__name",
+        )
+
+
+class FacilityDevice(AbstractBase):
+    """Facility devices ."""
+
+    facility = models.ForeignKey(Facility, on_delete=models.PROTECT)
+    number_of_devices = models.IntegerField(default=0)
```
Comment (kennedykori) [inline fahari/mle/models.py:44]: For ease of validation. Change this to a `models.PositiveSmallIntegerField`.

File: fahari/mle/models.py [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+
+class FacilityDeviceRequest(AbstractBase):
+    """Facility devices request by users."""
+
+    facility = models.ForeignKey(Facility, on_delete=models.PROTECT)
+    device_requested = models.CharField(max_length=50, default="")
+    NEW = "NEW"
+    REPLACEMENT = "REPLACEMENT"
+    REPAIR = "REPAIR"
+    REQUEST_TYPE_IDS = [(NEW, "New"), (REPLACEMENT, "Replacement"), (REPAIR, "Repair")]
```
Comment (kennedykori) [inline fahari/mle/models.py:67]: Use a choice enumeration type here too.

**Review by kennedykori — CHANGES_REQUESTED** [review]
Please fix those first.

File: fahari/mle/models.py (hunk 1 above)
Comment (saladgg) [reply fahari/mle/models.py:15]: fixed

File: fahari/mle/models.py (hunk 2 above)
Comment (saladgg) [reply fahari/mle/models.py]: fixed

File: fahari/mle/models.py (hunk 5 above)
Comment (saladgg) [reply fahari/mle/models.py:9]: Suggest a generic name that will tell whether a facility has internet and/or networked

File: fahari/mle/models.py (hunk 7 above)
Comment (saladgg) [reply fahari/mle/models.py]: fixed

File: fahari/mle/models.py (hunk 6 above)
Comment (saladgg) [reply fahari/mle/models.py:59]: fixed

File: fahari/mle/models.py (hunk 4 above)
Comment (saladgg) [reply fahari/mle/models.py:78]: device name. whether a server, desktop, orinter etc

File: fahari/mle/models.py (hunk 3 above)
Comment (saladgg) [reply fahari/mle/models.py]: fixed

**Pull request merged**